crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -52,13 +53,30 @@ impl Solution {
         score: SolverScore,
         weth: eth::WethAddress,
     ) -> Result<Self, SolutionError> {
+        let score_cip38 = SolverScoreCIP38 {
+            surplus: {
+                let mut surplus = vec![];
+                for trade in trades.iter() {
+                    match trade.score(&prices, weth) {
+                        Ok(score) => surplus.push(score),
```

Comment:
As is the `surplus` vector will contain multiple entries for the same token and 0 surplus entries for JIT orders.
Should we aggregate those surpluses in the same token and filter out 0 amounts to remove noise or are the individual entries specifically needed for some bookkeeping?

Reply:
Even if we aggregate them at the `solution` level, I have to do it again at the `settlement` level, since score is calculated for `settlement` and `settlement` can have multiple solutions. So i figured I just do it once at the `settlement` level and keep vector here for potential debugging purposes.